packages/contracts-bedrock/src/cannon/libraries/MIPSInstructions.sol

```diff
@@ -0,0 +1,265 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+/// @notice Execute an instruction.
+function executeMipsInstruction(uint32 insn, uint32 rs, uint32 rt, uint32 mem) pure returns (uint32 out) {
```

Comment:
Another question - I put this in the "libraries" directory, which I think follows existing conventions?  But this isn't actually a Solidity `library`.  Does this pattern make sense?

Reply:
We don't really have a style guide for this, but this is OK.